acceptancetests/clean_prodstack.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+import re
+import os
+import sys
+import logging
+import subprocess
+
+def get_current_controller_and_models():
+    lines = subprocess.check_output(
+        ['juju', 'models'], stderr=subprocess.STDOUT).split('\n')
```

Comment:
Use "-- format yaml" or "--format json"  and parse that into a datastructure, that you can then get the required details more easily.

For instance:
    model_output = json.loads(subprocess.check_output(['juju', 'models', '--format', 'json']))
    model_list = [m['name'] for m in  all_models['models']]
    controller_output = json.loads(subprocess.check_output(['juju', 'controllers', '--format', 'json']))
    controller_name = controller_output['current-controller']